From 359d63b62bd45ee657ba54506e0165888e73a3ef Mon Sep 17 00:00:00 2001
From: Joe Hirschfeld <j@ibj.io>
Date: Sat, 2 Dec 2023 14:38:27 -0800
Subject: [PATCH 6/6] Use system libdatachannel instead of the packaged version

---
 Makefile | 24 ++++--------------------
 1 file changed, 4 insertions(+), 20 deletions(-)

diff --git a/Makefile b/Makefile
index 732dab3..da7ea16 100644
--- a/Makefile
+++ b/Makefile
@@ -17,13 +17,11 @@ ifneq (x,x$(shell which ccache))
 CCACHE ?= ccache
 endif
 
-LIBDATACHANNEL_PATH ?= third_party/libdatachannel
-
 USE_HW_H264 ?= 1
 USE_FFMPEG ?= $(shell pkg-config libavutil libavformat libavcodec && echo 1)
 USE_LIBCAMERA ?= $(shell pkg-config libcamera && echo 1)
 USE_RTSP ?= $(shell pkg-config live555 && echo 1)
-USE_LIBDATACHANNEL ?= $(shell [ -e $(LIBDATACHANNEL_PATH)/CMakeLists.txt ] && echo 1)
+USE_LIBDATACHANNEL ?= $(shell pkg-config libdatachannel && echo 1)
 
 ifeq (1,$(DEBUG))
 CFLAGS += -g
@@ -49,14 +47,8 @@ LDLIBS += $(shell pkg-config --libs live555)
 endif
 
 ifeq (1,$(USE_LIBDATACHANNEL))
-CFLAGS += -DUSE_LIBDATACHANNEL
-CFLAGS += -I$(LIBDATACHANNEL_PATH)/include
-CFLAGS += -I$(LIBDATACHANNEL_PATH)/deps/json/include
-LDLIBS += -L$(LIBDATACHANNEL_PATH)/build -ldatachannel-static
-LDLIBS += -L$(LIBDATACHANNEL_PATH)/build/deps/usrsctp/usrsctplib -lusrsctp
-LDLIBS += -L$(LIBDATACHANNEL_PATH)/build/deps/libsrtp -lsrtp2
-LDLIBS += -L$(LIBDATACHANNEL_PATH)/build/deps/libjuice -ljuice-static
-LDLIBS += -lcrypto -lssl
+CFLAGS += -DUSE_LIBDATACHANNEL $(shell pkg-config --cflags libdatachannel)
+LDLIBS += $(shell pkg-config --libs libdatachannel)
 endif
 
 HTML_SRC = $(addsuffix .c,$(HTML))
@@ -68,10 +60,6 @@ all: version
 
 .SUFFIXES:
 
-ifeq (1,$(USE_LIBDATACHANNEL))
-camera-streamer: $(LIBDATACHANNEL_PATH)/build/libdatachannel-static.a
-endif
-
 .PHONY: version
 version:
 	printf "#define GIT_VERSION \"$(GIT_VERSION)\"\n#define GIT_REVISION \"$(GIT_REVISION)\"\n" > version.h
@@ -85,7 +73,7 @@ version:
 	$(CCACHE) $(CXX) $(CFLAGS) -o $@ $(filter-out cmd/%, $^) $(filter $</%, $^) $(LDLIBS)
 
 install: $(TARGET)
-	install $(TARGET) $(DESTDIR)/usr/local/bin/
+	install $(TARGET) $(DESTDIR)/bin/
 
 clean:
 	rm -f .depend $(OBJS) $(OBJS:.o=.d) $(HTML_SRC) $(TARGET)
@@ -105,7 +93,3 @@ headers:
 html/%.c: html/%
 	xxd -i $< > $@.tmp
 	mv $@.tmp $@
-
-$(LIBDATACHANNEL_PATH)/build/libdatachannel-static.a: $(LIBDATACHANNEL_PATH)
-	[ -e $</build/Makefile ] || cmake -S $< -B $</build
-	$(MAKE) -C $</build datachannel-static
-- 
2.39.3 (Apple Git-145)

